### PR TITLE
Update Secure Messaging authorization rules

### PR DESCRIPTION
**Pull request by dcloud**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- Added `verify` to `RequiredLoginView` ([when a page requires a signed-in, identity-verified account](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/digital-health-modernization/engineering/mhv-authorization.md#is-signed-in-is-identity-verified))
- Changed redirect to point the MHV-on-VA.gov Landing page, `/my-health`.

Per [Tom Black from OCTO identity team](https://dsva.slack.com/archives/C0581MN69TJ/p1717427161772569?thread_ts=1717185477.450519&cid=C0581MN69TJ):

> Re: identity proofing policies, we have several in place (and more on the way with new NIST guidance). The only thing that’s important really for VA applications is if the account is verified, rather than the specific policy that was used to verify the individual. We’ll have more guidance and updated documentation for VFS teams in the near future. 

See also  [prior discussion in #mhv-secure-messaging channel](https://dsva.slack.com/archives/C03ECSBGSKX/p1719417514868069)

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/86422

## Testing done

- Prior to the change, users with without `messaging` in their services list were redirected to an informational page, now they are redirected to the MHV-on-VA.gov Landing page
- Before, unverified users were not redirected to the verify page. Now, using [RequiredLoginView's verify rule](https://github.com/department-of-veterans-affairs/vets-website/blob/f9700b5f87d57b95452be70e154f25ba6dc1975d/src/platform/user/authorization/components/RequiredLoginView.jsx#L77), unverified users will get redirected to the verify page

## What areas of the site does it impact?

MHV Secure Messaging, and (indirectly) the MHV-on-VA.gov landing page

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv-secure-messaging/tests/containers/App.unit.spec.jsx
Comment (dcloud) [inline src/applications/mhv-secure-messaging/tests/containers/App.unit.spec.jsx:1]: Looks like this file was using windows <CR><LF> endings…

**Comment by dcloud** [comment]
Added context concerning identity checks:

> Re: identity proofing policies, we have several in place (and more on the way with new NIST guidance). The only thing that’s important really for VA applications is if the account is verified, rather than the specific policy that was used to verify the individual. We’ll have more guidance and updated documentation for VFS teams in the near future. 

See link in description summary above

**Review by alex-morgun-1 — DISMISSED** [review]

**Review by alex-morgun-1 — APPROVED** [review]

File: src/applications/mhv-secure-messaging/containers/App.jsx [hunk 1]
```diff
@@ -143,10 +143,11 @@ const App = ({ isPilot }) => {
     <RequiredLoginView
       user={user}
       serviceRequired={[backendServices.MESSAGING]}
+      verify
     >
       {user.login.currentlyLoggedIn &&
       !userServices.includes(backendServices.MESSAGING) ? (
-        window.location.replace('/health-care/secure-messaging')
+        window.location.replace('/my-health')
```
Comment (carlosfelix2) [inline src/applications/mhv-secure-messaging/containers/App.jsx:150]: Why not use the same Paths constant here?

**Review by carlosfelix2 — APPROVED** [review]

File: src/applications/mhv-secure-messaging/containers/App.jsx (hunk 1 above)
Comment (dcloud) [reply src/applications/mhv-secure-messaging/containers/App.jsx:150]: I only noticed the `Paths` constant is in the e2e tests directory! Not sure why they weren't used here. Maybe I'll do a follow-up PR

**Pull request merged**